### DISCUSSION TITLE
Error when set_state_simple() used

### DISCUSSION THREAD

**Discussion by hovnatan** [NVIDIA/cuQuantum#158] [Q&A] [answered]
When I run the following code with MPI: `mpirun -np 2`

    circuit = qiskit.circuit.library.QuantumVolume(n_qubits, depth=5, seed=1001)
    ket = np.zeros((2**n_qubits, 1))
    ket[2] = 1
    circuit.set_state_simple(ket)
    circuit.measure_all()
    circuit = transpile(circuit, simulator)
    simulator = StatevectorSimulator()
    job = simulator.run(circuit, shots=1024)

I get the following error: `cusvaer.backends.ubackendext.RuntimeError: stateSize does not match with the state vector size`

If I omit `circuit.set_state_simple(ket)` the simulation works. Can you please help?

**Comment by ymagchi** [comment, marked as answer]
Hi @hovnatan,

For the distributed state vector simulation in cusvaer, `set_state_simple()` sets 1 sub state vector assigned to each process: https://docs.nvidia.com/cuda/cuquantum/latest/appliance/cusvaer.html#set-state-simple-state.
In the above case, the expected state size will be `(2**n_qubits) / 2`.
